### ISSUE TITLE
Unable to run in the org mode source code block

### ISSUE THREAD

**Issue by kkweon** [prettier/prettier-emacs#38]
## Steps to reproduce

1. Create any org file `temp.org`
2. Create a source code block

```org
#+BEGIN_SRC js
  const a = {
    data: [],
    nextOffset: 1
  }
#+END_SRC
```
3. Edit and run `prettier-js`

```
prettier errors:
[error] No parser could be inferred for file: /Users/kkweon/temp/temp.org[*Org Src temp.org[ js ]*]
```

Thank you.

**Comment by rcoedo** [comment]
You could fix this using the custom variable `prettier-js-args` to specify the prettier parse you want to use.

**Comment by kkweon** [comment]
And can still be used to format js, ts, json, less, scss ? It will force the parser?

**Comment by rcoedo** [comment]
Yes, it forces the parser. You will have to set the variable to the right format for each one of these files. You could do this with hooks for example.

**Comment by bcc32** [comment]
Sorry to revive an old thread, but perhaps `prettier` could infer a reasonable extension from the major mode when there is no `buffer-file-name`, as is the case in org-mode source blocks?

**Comment by jacksonrayhamilton** [comment]
> Sorry to revive an old thread, but perhaps `prettier` could infer a reasonable extension from the major mode when there is no `buffer-file-name`, as is the case in org-mode source blocks?

I like this idea.

I think we might also need a new function `prettier-js-prettify-region`, to format only the block.  Currently the only way to format is the whole buffer.

We could also provide another new function, `prettier-js-prettify-code-block`, which, when called interactively with point inside a code block, could infer a file extension for the block, and figure out the beginning and end of that block to pass as args to `prettier-js-prettify-region`.
